src/main/kotlin/com/cultureamp/eventsourcing/BatchedAsyncEventProcessor.kt

```diff
@@ -66,7 +66,14 @@ class BatchedAsyncEventProcessor<M : EventMetadata>(
     )
 
     fun processOneBatch(): Action {
-        val startBookmark = bookmarkStore.bookmarkFor(bookmarkName)
+        val startBookmark = bookmarkStore.checkoutBookmark(bookmarkName).let {
+            if (it is Left<LockNotObtained>)
+                // try again shortly
+                // TODO should we log this?
```

Comment:
I reckon add a log hook like startLog and endLog - we default it to no-op, or maybe it has a sensible default of always logging?